src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,172 @@
+
+## Adding some "hot wires" to the equation
+
+We've collected more than a hundred responses from Action Cable / AnyCable users around the world, and found that a good portion of them build applications on top of [Hotwire][hotwire] or [Stimulus Reflex][sr] (+ [CableReady][cr]). Which is pretty cool: [_frontendless_ Rails frontend][frontendless-rails] is getting more and more traction!
+
+What can AnyCable give to these users? Let's take a look at how Turbo Streams work with Rails.
+
+We have the [turbo-rails][] gem, which integrate Action Cable with Hotwire. All you need is to drop a helper into your template to start consuming streams:
+
+```erb
+<%= turbo_stream_from some_model %>
+```
+
+The helper adds a [`<turbo-cable-stream-source>` element][turbo-cable-stream-source], which initiates an Action Cable subscription. The corresponding server-side code is as follows:
+
+```ruby
+class Turbo::StreamsChannel < ActionCable::Channel::Base
+  extend Turbo::Streams::Broadcasts, Turbo::Streams::StreamName
+
+  def subscribed
+    if verified_stream_name = self.class.verified_stream_name(params[:signed_stream_name])
+      stream_from verified_stream_name
+    else
+      reject
+    end
+  end
+end
+```
+
+The `#verified_stream_name` method uses an [`ActiveSupport::MessageVerifier`][message-verifier] to decode and verify the stream name. The subscription is rejected if the signed stream name is invalid.
+
+The code above is structurely the same as the one we wrote for JWT identification. What does it mean? We can "move" the implementation to AnyCable Go server and **create subscriptions without touching RPC**. Again.
+
+That's exactly what we did as a part of the [Signed streams][signed-streams-docs] feature. And we did it not only for Turbo Streams, but for Cable Ready as well (since its [`#stream_from` implementation][cr-stream-from] is almost the same).
+
+> Combining JWT identification with signed streams, it is possible to avoid running RPC server at all in case you only use Hotwire/CableReady functionality.
+
+Below is the visualization of the RPC server metrics during Hotwire benchmarks with four different configuration (blue markers), from left to rifght: baseline (AnyCable PRO w/o any features enabled), with JWT identification, with signed streams, and with both features enabled at the same time.
+
+<figure class="blog--figure">
+  <img class="blog--media" title="AnyCable metrics when running with and without JWT and 'hot' streams" src="./grafana.png" width="1364" height="624">
+</figure>
+
+No surpises that the last one shows zeros for RPC metrics—it hasn't been bothered at all. We can also notice that the AnyCable Go CPU usage is the lowest for the fourth scenario—**dealing with tokens and signatures is "cheaper" than performing gRPC calls**.
```

Comment:
It shouldn't come as any surprise that the last one shows zeros for RPC metrics—it wasn't impacted at all. We can also note that CPU usage with AnyCable Go is the lowest in the fourth scenario—**dealing with tokens and signatures is "cheaper" than performing gRPC calls**.